apps/server/src/modules/files-storage/controller/api-test/files-storage-preview.api.spec.ts

```diff
@@ -0,0 +1,437 @@
+			status: response.status,
+		};
+	}
+
+	async postUploadFromUrl(routeName: string, data: Record<string, unknown>) {
+		const response = await request(this.app.getHttpServer()).post(routeName).send(data);
+
+		return {
+			result: response.body as FileRecordResponse,
+			error: response.body as ApiValidationError,
+			status: response.status,
+		};
+	}
+
+	async getDownloadFile(routeName: string, query?: string | Record<string, unknown>) {
+		const response = await request(this.app.getHttpServer())
+			.get(routeName)
+			.query(query || {});
+
+		return {
+			result: response.body as FileRecordResponse,
+			error: response.body as ApiValidationError,
+			status: response.status,
+		};
+	}
+
+	async getDownloadFileBytesRange(routeName: string, bytesRange: string, query?: string | Record<string, unknown>) {
+		const response = await request(this.app.getHttpServer())
+			.get(routeName)
+			.set('Range', bytesRange)
+			.query(query || {});
+
+		return {
+			result: response.body as FileRecordResponse,
+			error: response.body as ApiValidationError,
+			status: response.status,
+			headers: response.headers as Record<string, string>,
+		};
+	}
+}
+
+const createRndInt = (max) => Math.floor(Math.random() * max);
+
+describe('File Controller (API) - preview', () => {
+	let module: TestingModule;
+	let app: INestApplication;
+	let em: EntityManager;
+	let s3ClientAdapter: DeepMocked<S3ClientAdapter>;
+	let currentUser: ICurrentUser;
+	let api: API;
+	let validId: EntityId;
+	let appPort: number;
+
+	beforeAll(async () => {
+		appPort = 10000 + createRndInt(10000);
+
+		module = await Test.createTestingModule({
+			imports: [FilesStorageTestModule],
+		})
+			.overrideProvider(AntivirusService)
+			.useValue(createMock<AntivirusService>())
+			.overrideProvider(S3ClientAdapter)
+			.useValue(createMock<S3ClientAdapter>())
+			.overrideGuard(JwtAuthGuard)
+			.useValue({
+				canActivate(context: ExecutionContext) {
+					const req: Request = context.switchToHttp().getRequest();
+					req.user = currentUser;
+					return true;
+				},
+			})
+			.compile();
+
+		app = module.createNestApplication();
+		const a = await app.init();
+		await a.listen(appPort);
+
+		em = module.get(EntityManager);
+		s3ClientAdapter = module.get(S3ClientAdapter);
+		api = new API(app);
+	});
+
+	afterAll(async () => {
+		await app.close();
+		await module.close();
+	});
+
+	beforeEach(async () => {
+		jest.resetAllMocks();
+		await cleanupCollections(em);
+		const school = schoolFactory.build();
+		const roles = roleFactory.buildList(1, {
+			permissions: [Permission.FILESTORAGE_CREATE, Permission.FILESTORAGE_VIEW],
+		});
+		const user = userFactory.build({ school, roles });
+
+		await em.persistAndFlush([user, school]);
+		em.clear();
+		validId = school.id;
+		currentUser = mapUserToCurrentUser(user);
+	});
+
+	describe('preview', () => {
+		describe('with bad request data', () => {
+			it('should return status 400 for invalid recordId', async () => {
```

Comment:
We don't follow the scenario structure with setup and so on in the other api tests. But it's basically a requirement for all tests. Should we also use this structure here?